docs/asyncapi-file.md

```diff
@@ -1,6 +1,98 @@
+
+In the following sections, you'll learn about the inner working of the generator, what happens once the specification file is fed to the generator, and how you can use the content of the specification file with either an AsyncAPI document or AsyncAPI string in your template.
+
+## The generation process
+1. The **Generator** receives the **AsyncAPI file** as an input. 
+2. The Generator sends to the **[Parser](parser.md)** a stringified version of the original specification file to validate the JSON/YAML **AsyncAPI file**.
+3. The **parser** validates the **AsyncAPI file** using either the OpenAPI, RAML, or Avro schemas defined in the specification files. 
+4. If the **parser** determines that the **AsyncAPI file** is valid, it manipulates the original JSON/YAML specification file into functions and properties, bundling them together into an **asyncapi**(asyncapiDocument). 
+5. At this point, the **Generator** passes the **originalAsyncAPI** and the **asyncapi** which make up part of the **[template context](asyncapi-context.md)** to the **render engine**. 
+6. The **template context** is accessible to the template files that are passed to either the [react](react-render-engine.md) or [nunjucks](nunjucks-render-engine.md) **render engines** by the **Generator**.
+   
+``` mermaid
+graph LR
+    A[Template Context]
+    B{Generator}
+    C[Parser]
+    D[Render Engine]
+    E[AsyncAPI File] --> B
+  subgraph Generator
+    B -->| asyncapiString | C
+    C --> | asyncapi -> AsyncAPIDocument type | A
+    B--> | originalAsyncAPI -> Stringified document | A
+    A --> D
+  end
+  ```
+The AsyncAPI file's content is accessible to you while writing your template in two distinct ways:
+1. The `originalAsyncAPI`, which is a stringified version of the AsyncAPI file provided as input, without any modifications.
+2. The `asyncapi` (`AsyncAPIDocument`) which is an object with a set of functions and properties, that comes as a result of the AsyncAPI Parser manipulating the `originalAyncAPI` .The `asyncapi` methods and parameters make it easier to access contents of AsyncAPI file in your templates.
+
+> Check out the structure of the [AsyncAPIDocument](https://github.com/asyncapi/parser-js/blob/master/API.md#module_@asyncapi/parser+AsyncAPIDocument)
+
+In the following sections, you will learn how to use either the **originalAsyncAPI** or the **asyncapi** in your template.
+
+### Method 1: `originalAsyncAPI` and template 
+One way of using the contents of AsyncAPI file inside your template files is by using its stringified version that reflects exactly the same contents as the AsyncAPI file provided as an input to the AsyncAPI Generator CLI. You can access this stringified AsyncAPI file directly in your templates using the `originalAsyncAPI` variable. You also access it via the [hooks](hooks.md) `generator.originalAsyncAPI` functionality because `originalAsyncAPI` is also a part of the Generator instance that is passed to hooks.
```

Comment:
```suggestion
One way of using the contents of AsyncAPI document inside your template files is by using its stringified version that reflects exactly the same contents as the AsyncAPI document provided as an input to the Generator. You can access it directly in your templates using the `originalAsyncAPI` variable. You also access it via the [hooks](hooks.md) `generator.originalAsyncAPI` because `originalAsyncAPI` is also a part of the Generator instance that is passed to hooks.
```